<?xml version="1.0"?>
<launch>

    <arg name="follower"    default="pure_pursuit"  doc="pure_pursuit, stanley" />
    
    <group ns="control">

        <!-- Remap topics -->
        <remap from="current_pose"     to="/localization/current_pose" />
        <remap from="current_velocity" to="/localization/current_velocity" />
        <remap from="path"             to="/planning/path" />


        <!-- Follower -->
        <node if="$(eval arg('follower')=='pure_pursuit')" pkg="autoware_mini" type="pure_pursuit_follower.py" name="pure_pursuit_follower" output="screen" required="true" />

        <node if="$(eval arg('follower')=='stanley')" pkg="autoware_mini" type="stanley_follower.py" name="stanley_follower" output="screen" required="true" />

        <rosparam command="load" file="$(find autoware_mini)/config/control.yaml"/>


        <!-- convert twist to carla ackermann drive-->
        <node pkg='autoware_mini' type='vehiclecmd_to_ackermanndrive' name='vehiclecmd_to_ackermanndrive' output='screen'>
            <!-- <param name="role_name" value="$(arg role_name)"/> -->
        </node>

        <node pkg='autoware_mini' type='carla_to_autoware_vehicle_status' name='carla_to_autoware_vehicle_status' output='screen'>
            <!-- <param name="role_name" value="$(arg role_name)"/> -->
        </node>

    </group>

</launch>